syntax = "proto3";

package mserialize;

message LogMessage{
	string content = 1;
}

message State{

    enum StateType{
        READY = 0;
        BUSY = 1;
        KILLED = 2;
    }

    int32 id = 1; // state ID
    StateType type = 2; // Type of state
    string reason = 3; // Reason for execution stopping
    int32 num_executing = 4; // number of executing instructions
    int32 wait_time = 5; 
}

message StateList{
    repeated State states = 1; 
}

message MessageList{
    repeated LogMessage messages = 1; 
}
